packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2405,3 +2406,45 @@ const waitForSeq = async (
     };
     deltaManager.on("op", handleOp);
 });
+
+export interface ISummarizerRequestOptions {
+    cache: boolean,
+    reconnect: boolean,
+    summarizingClient: boolean,
+    url: string,
+}
+
+/**
+ * Requests a Summarizer.
+ * @param loader - the loader that resolves the request
+ * @param lastSequenceNumber - the last sequence number (e.g., from DeltaManager)
+ * @param cache - use cache to retrieve summarizer
+ * @param summarizingClient - is summarizer client
+ * @param reconnect - can reconnect on connection loss
+ */
+export async function requestSummarizer(
+    loader: ILoader,
+    lastSequenceNumber: number,
+    { cache, reconnect, summarizingClient, url }: ISummarizerRequestOptions): Promise<ISummarizer> {
```

Comment:
Any reason url is part of ISummarizerRequestOptions? And not hard coded in this function?
It's fixed (if we are looking for summarizer, not something else).
So it feels like there is no reason for external code needing to know that URL

Reply:
Note that any other URI other than "/" would not work, because "/" is special - it is processed by relative loader installed by Contaner - it short circuits it to current container. Anything else will go up the stack and would not resolve properly unless it has full container path.

Reply:
This method can also be called directly by a user who is not referring to a current container. That answers the requirement that you and Tony spoke about regarding the ability to create a summarizer without a current container.

The test does this:
    const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
    await container.attach(provider.driver.createCreateNewRequest(provider.documentId));
    const absoluteUrl = await container.getAbsoluteUrl("");
    if (absoluteUrl === undefined) {
        throw new Error("URL could not be resolved");
    }
    options.url = absoluteUrl;
    return requestSummarizer(loader, lastSequenceNumber, options);

Reply:
Got it. I understand the need, but I'll point out that there is discrepancy here. If we use the loader from container runtime, then it has relative loader adapter in the chain and has special handling for "/".
And if it's "regular" loader, then "/" would fail.
This is absolutely not clear from API shape and developers will get confused.

So, I think we either want to say that this API is to be used with loader from container runtime and there is no URI, or we need to revisit usage of relative loader (remove it?) and force paths like that to always provide proper absolute path (container runtime to provide it - it already does through getAbsoluteUrl().

@anthony-murphy - what do you think?

Reply:
I merged the create-elected-summarizer and create-on-demand-summarizer flow into one function, because the logic was so similar, but I could easily split the two for clarity. As you pointed out elsewhere, the ContainerRuntime loader shouldn't be exposed, so the create-on-demand flow should really always get an absolute URL. It could optionally fail on URL beginning with "/".